### PR TITLE
Sex from chi

### LINKED ISSUE #43: Function to extract sex from CHI
Related to the other issue I just added #42 but much simpler as there's no ambiguity with sex in the CHI number.

[UoE implementation](https://github.com/SurgicalInformatics/chiTools/blob/master/R/chi.R)
[My implementation](https://github.com/Health-SocialCare-Scotland/cocin-scotland/blob/master/functions/fix_age_sex_from_chi.R) for the data I'm working on there it returns a factor. I think default behaviour should be to return 1/2 with maybe an option to return as a factor.

### PR DESCRIPTION
**Pull request by Moohan**
Opening as draft pending comments in the issue.
Closes #43

### CODE REVIEW

**Review by 080708km — CHANGES_REQUESTED** [review]
Delete

**Comment by davidc92** [comment]
Hi James

I'm just getting round to looking at this now. Before I do a review I'd just like to understand the scope for this function a bit more because I'm not entirely convinced this is all that useful? The reason being, I don't know of any dataset that has chi on it that does not also have sex. I also have some questions around how accurate this is - how are people who are intersex or their sex is otherwise unknown accounted for here? 

D

**Review by davidc92 — CHANGES_REQUESTED** [review]
Just a few minor comments. While you're updating, just in the interest of brevity, can you re-knit the readme.rmd file so that the .md is in sync as part of this PR please? D

File: DESCRIPTION [hunk 1]
```diff
@@ -11,7 +11,8 @@ Authors@R: c(
     person("Jaime", "Villacampa", email = "jaime.villacampa@nhs.net", role = "aut"),
     person("Graeme", "Gowans", email = "graeme.gowans@nhs.net", role = "aut"),
     person("Alice", "Byers", email = "alice.byers@nhs.net", role = "ctb"),
-    person("Alan", "Yeung", email = "alan.yeung@nhs.net", role = "ctb")
+    person("Alan", "Yeung", email = "alan.yeung@nhs.net", role = "ctb"),
+    person("James", "McMahon", email = "j.mcmahon1@nhs.net", role = "ctb")
```
[the remark this answers is not in the document]
Comment (Moohan) [reply DESCRIPTION]: Done

File: R/chi.R [hunk 2]
```diff
@@ -151,3 +151,88 @@ chi_pad <- function(x) {
...
+#' even for women.
+#'
+#' The default behaviour for \code{sex_from_chi} is to first check the CHI number is
+#' valid using \code{check_chi} and then to return 1 for male and 2 for female.
+#'
+#' There are options to return custom values e.g. \code{'M'} and \code{'F'} or to return
+#' a factor which will have lables \code{'Male'} and \code{'Female')}
+#'
+#' @param chi_number a CHI number or a vector of CHI numbers with \code{character} class.
+#' @param male_value,female_value optionally supply custom values for Male and Female. Note
```
[the remark this answers is not in the document]
Comment (Moohan) [reply R/chi.R:176]: Not done - no space means that that note applies to both of the parameters. Check the doc with `?sex_from_chi` and you'll see what I mean.

File: R/chi.R [hunk 3]
```diff
@@ -151,3 +151,88 @@ chi_pad <- function(x) {
...
+#' that that these must be of the same class.
+#' @param as_factor logical, optionally return as a factor with lables \code{'Male'}
```
[the remark this answers is not in the document]
Comment (Moohan) [reply R/chi.R:178]: Done

**Comment by Moohan** [comment]
Made most of the changes, see comment above.

I updated the `readme.rmd` and re-knitted, I got the same error again about proxy (#55) ... However, the `.md` file still updated and looks ok to me (both in markdown and if you look at it on GitHub for this branch). There were also merge conflicts for some reason which I resolved.

I also just added a short bit to the readme for this function, I'm pretty happy that they're in sync now.

**Review by davidc92 — APPROVED** [review]

**Pull request merged**